runtime/src/contains.rs

```diff
@@ -0,0 +1,49 @@
+use std::{
+    borrow::Borrow,
+    cmp::Eq,
+    collections::{HashMap, HashSet},
+    hash::Hash,
+};
+
+pub trait Contains<'a, T: Eq + Hash> {
+    type Item: Borrow<T>;
+    type Iter: Iterator<Item = Self::Item>;
+    fn contains(&self, key: &T) -> bool;
+    fn contains_iter(&'a self) -> Self::Iter;
+}
+
+impl<'a, T: 'a + Eq + Hash, U: 'a> Contains<'a, T> for HashMap<T, U> {
+    type Item = &'a T;
+    type Iter = std::collections::hash_map::Keys<'a, T, U>;
+
+    fn contains(&self, key: &T) -> bool {
+        self.contains_key(key)
```

Comment:
how about using fully qualified syntax like this for consisntent like `<HashSet<T>>::contains(self, key)`?